### DISCUSSION TITLE
Help in doing an RPC call through RESTCONF?

### DISCUSSION THREAD

**Discussion by rFronteddu** [freeconf/restconf#24] [General]
I am trying to use curl to call the rotateTire method. The yang and the code are the same as your example, I tried the following and other variations:
````
$ curl -X POST http://localhost:8080/restconf/data/car:rotateTires
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    12  100    12    0     0  12752      0 --:--:-- --:--:-- --:--:-- 12000Bad Request

$ curl -d "" -X POST http://localhost:8080/restconf/data/car/rotateTires
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   194  100   194    0     0   171k      0 --:--:-- --:--:-- --:--:--  189k{"ietf-restconf:errors":{"error":[{"error-type":"protocol","error-tag":"operation-failed","error-path":"/restconf/data/car/rotateTires","error-message":"not found. no module found in path"}
]}}
```
Can you help me figure out what am I doing wrong?

**Comment by dhubler** [comment]
probably nothing, i haven't updated the examples in a while and will get
you an answer in the next hour or so.

**Comment by rFronteddu** [comment]
Thanks! Let me know if you want to see my code or the yang model.

**Comment by dhubler** [comment]
wow, my bad, i introduced a bug when adding strict compliance.  I updated
restconf master w/fix.  you will need to update your version of examples.
I had outstanding changes so hopefully it don't disrupt you too much

**Comment by rFronteddu** [comment]
Thanks! I just started working on this project and experimenting with what stack we could use so it is the perfect time for disruptive changes :)

**Comment by rFronteddu** [comment]
That fixed it, I am now able to do RPC calls.

I have another question if I set config false in a leaf, does that protect the leaf from being updated without using an RPC call?

**Comment by dhubler** [comment]
re:fix
nice

re: config false
correct if i understand you correctly.  Update cannot be attempted via RPC.
https://freeconf.org/docs/reference/yang-primer/#metrics

**Reply by rFronteddu** [reply]
You mean without right?  I think in your example you called reset through RPC to set miles to 0. While you changed the speed with a put.

**Comment by dhubler** [comment]
config false means a PUT or POST to that to that field  **directly** is not
possible

rpcs do whatever you want them too, update, clear, read so you can
**indirectly** change anything I suppose, but that is not really what they
were intended for.

**Reply by rFronteddu** [reply]
I was thinking of using RPC functionalities like that because it looked like an easy way to pair incremental changes with sanity checks. 

It is really not a configuration that I am changing but the dynamic state of a component. I may be abusing RESTCONF a bit. I am just playing around for now. I will check your other examples to see how you implemented notifications and other features. There may be a more idiomatic way of doing what I am attempting.

**Comment by dhubler** [comment]
Yep it will become clear after some playing around.
